### PR TITLE
Improvements to stopping systemd service

### PR DESCRIPTION
**Pull request by nre-ableton**
This PR fixes a number of issues with how the systemd service was being stopped, namely:

- Fixes an issue where the PID file would contain an extra line for the `sh -c` PID
- Only attempt to stop the service if it is actually running (which allows us to remove a `ignore_errors: true` attribute)
- Improved task to wait for Jenkins to actually stop
- Sync filesystems after shutting down Jenkins

### CODE REVIEW

**Comment by ala-ableton** [comment]
@nre-ableton I'm interested in reviewing this PR, thanks!

**Review by rco-ableton — APPROVED** [review]

**Comment by nre-ableton** [comment]
I pushed a [fixup](https://github.com/Ableton/ansible-role-jenkins-jcasc/pull/167/commits/c0929f9430296ecc583372399cb9724e0cb83dba) as a workaround for https://github.com/ansible-community/ansible-lint/issues/1894.

File: templates/jenkins.service.j2 [hunk 1]
```diff
@@ -7,10 +7,8 @@ Environment="CASC_JENKINS_CONFIG={{ jenkins_config_dir }}/jenkins.yaml"
 Environment="JENKINS_HOME={{ jenkins_home }}"
 ExecStart={{ jenkins_command }}
 # So systemd won't generate a PID file for non-forking services, so we need to make one.
-# Here we grep only for the WAR filename, not the whole process name, or else quotes and
```
Comment (ala-ableton) [inline templates/jenkins.service.j2:10]: > While pgrep guarantees that its root process won't be matched, this does
not apply to the root process of that process,

Why are you talking about "root process"? That seems out of topic.

File: tasks/stop-service.yml [hunk 2]
```diff
@@ -18,9 +23,17 @@
       # https://github.com/ansible-community/molecule/issues/2765
       changed_when: false
 
-    - name: Wait for Jenkins to stop  # noqa: no-handler
-      wait_for:
-        timeout: 5
+    # Just to be safe, wait until Jenkins finally stops. Stopping services can be
+    # asynchronous and so the process itself may continue to run for a bit to ensure a
+    # clean shut-down. We suspect that Jenkins cleanup routines run in this way may be the
+    # source of mysterious "file has vanished" errors from rsync.
```
Comment (ala-ableton) [inline tasks/stop-service.yml:29]: I don't see why we mention rsync here.

File: tasks/stop-service.yml [hunk 3]
```diff
@@ -34,6 +34,11 @@
       until: jenkins_service_running is failed
       retries: 60
       delay: 1
+
+    # Again, a paranoid measure aimed at preventing potential rsync failures.
```
Comment (ala-ableton) [inline tasks/stop-service.yml:38]: Similarly, I don't see why we mention rsync here.

File: templates/jenkins.service.j2 (hunk 1 above)
Comment (nre-ableton) [reply templates/jenkins.service.j2:10]: I've reworded the commit message to be a bit clearer. See c107143 (and sorry for the force-push).

File: tasks/stop-service.yml (hunk 2 above)
Comment (nre-ableton) [reply tasks/stop-service.yml]: Fixed in 6dfaba6.

File: tasks/stop-service.yml (hunk 3 above)
Comment (nre-ableton) [reply tasks/stop-service.yml]: Fixed in 7d8ade8.

File: tasks/stop-service.yml [hunk 4]
```diff
@@ -25,8 +25,8 @@
 
     # Just to be safe, wait until Jenkins finally stops. Stopping services can be
     # asynchronous and so the process itself may continue to run for a bit to ensure a
-    # clean shut-down. We suspect that Jenkins cleanup routines run in this way may be the
-    # source of mysterious "file has vanished" errors from rsync.
+    # clean shut-down. If subsequent tasks must od something with files in the Jenkins
```
Comment (ala-ableton) [inline tasks/stop-service.yml:28]:
```suggestion
    # clean shut-down. If subsequent tasks must do something with files in the Jenkins
```

File: templates/jenkins.service.j2 [hunk 5]
```diff
@@ -7,10 +7,8 @@ Environment="CASC_JENKINS_CONFIG={{ jenkins_config_dir }}/jenkins.yaml"
...
-# other symbols in the command will cause pgrep to fail. Also, systemd doesn't allow
-# redirection, so we need to wrap the command with sh.
-ExecStartPost=sh -c 'pgrep -f {{ jenkins_war }} > {{ jenkins_pid_file }}'
+# Also, systemd doesn't allow redirection, so we need to wrap the command with sh.
+ExecStartPost=sh -c 'systemctl show jenkins -p ExecMainPID | cut -d = -f 2 > {{ jenkins_pid_file }}'
```
Comment (ala-ableton) [inline templates/jenkins.service.j2:11]: In c1071432a62fc21ff9ff250376e84309fd881655 's commits message: "pgrep would returning two" -> "pgrep was returning two"

**Review by ala-ableton — COMMENTED** [review]
Just one minor comment

File: templates/jenkins.service.j2 (hunk 5 above)
Comment (nre-ableton) [reply templates/jenkins.service.j2:11]: I think I meant "would return two". See bfb55df.

**Review by ala-ableton — APPROVED** [review]

**Pull request merged**